build.gradle

```diff
@@ -315,42 +225,44 @@ subprojects {
 
         dependencies {
             configurations.all {
-                // Kotlin compiler classpaths don't support BOM nor need it.
-                if (it.name.endsWith('Classpath') && !it.name.startsWith('kotlin')) {
-                    boms.each {bom ->
-                        add(it.name, enforcedPlatform(bom))
-                    }
-                }
+                // Gradle and newer plugins will set these configuration properties correctly.
+                if (it.canBeResolved && !it.canBeConsumed
```

Comment:
This looks like a workaround.

Reply:
The below line is the workaround :) This line is the recommendation I got from Gradle team a while ago on how to apply dependency management with platforms.